test/override-types.test-d.ts

```diff
@@ -0,0 +1,125 @@
+import { expectType } from 'tsd'
+import { TypeEqual } from 'ts-expect'
+import { PostgrestClient } from '../src'
+import { CustomUserDataType, Database } from './types'
+
+const REST_URL = 'http://localhost:54321'
+const postgrest = new PostgrestClient<Database>(REST_URL)
+
+// Test merge array result to object should error
+{
+  const singleResult = await postgrest
+    .from('users')
+    .select()
+    .overrideTypes<{ custom_field: string }>()
+  if (singleResult.error) {
+    throw new Error(singleResult.error.message)
+  }
+  let result: typeof singleResult.data
+  expectType<
+    TypeEqual<
+      typeof result,
+      {
+        Error: 'Type mismatch: Cannot cast array result to a single object. Use .returns<Array<YourType>> for array results or .single() to convert the result to a single object'
```

Comment:
Should this point to `overrideTypes` instead of `returns`?

Reply:
Good catch